### DISCUSSION TITLE
How to properly debug?

### DISCUSSION THREAD

**Discussion by Thiritin** [telegraf/telegraf#1249] [Q&A] [answered]
Hey, I was playing around with the Framework for a while. When I made an error I received just "TypeError" message with no proper stack trace or answer to what the error could have been.

Is there a good solution on how to develop with this tool?

**Comment by wojpawlik** [comment, marked as answer]
In general, `export DEBUG='telegraf:*` before starting the bot.

If you got `UnhandledPromiseRejectionWarning`, as a workaround run `node` with  [`--trace-warnings`,](https://nodejs.org/api/cli.html#cli_trace_warnings) flag. For a proper fix, use https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md to flag incorrect `Promise` usage which bypasses Telegraf error handling.

**Comment by tassoman** [comment]
Looks like using `DEBUG='telegraf:*' nodemon bot.js`, queries aren't sent to the Telegram apis 🤔
